Adjust open id provider login

larissazech opened the pull request:
Use example client to test the whole flow and decrypt the id_token containing the chosen username.

Since this branch was forked from add_open_id_provider, it might be necessary to merge PR #41 first and to rebase this one on master afterwards.

corresponding issue: #34

auvin committed:
added oidc-provider to api container

auvin committed:
Added OIDC-Provider as app

  * added oidc-provider to installed apps
  * added templates for oidc pages
  * configured urls
  * added OIDC section to README

  * Added e2e test
    * Created dango-admin commands 'createuser', 'registerclient', 'deleteuser' and 'deleteallclients'
      * created admin_commands module for management commands
      * adjusted travis script to create example client and delete it after testruns
    * e2e tests
      * availability of oidc-urls
      * authorization process of an example client

auvin committed:
added templates and configured urls

auvin committed:
added templates and configured urls

auvin committed:
add login url for correct redirect to /authorize

larissazech committed:
Adjust openID-provider login-page for use with eID-client-checker

Let openID-provider login-url point to eID-client-checker.

Let django manage static content in checker for being able to load dynamic content. Therefore, the clear-flag in entrypoint.sh has to be removed.

Temporarily include skip-button in checker for sending POST to provider. After entering a username and pressing the button, the user is logged in and a valid session is started. Afterwards, a redirect to the client occurs.

Resolves: #34

larissazech committed:
Adjust openID-provider login-page for use with eID-client-checker

Let openID-provider login-url point to eID-client-checker.

Let django manage static content in checker for being able to load dynamic content. Therefore, the clear-flag in entrypoint.sh has to be removed.

Temporarily include skip-button in checker for sending POST to provider. After entering a username and pressing the button, the user is logged in and a valid session is started. Afterwards, a redirect to the client occurs.

Resolves: #34

nils-wisiol commented on api/boilerplate/templates/check/check.html:
```diff
@@ -0,0 +1,34 @@
+<!doctype html>
```
please fix indentation in this file (according to #30 )

nils-wisiol commented on api/boilerplate/views.py:
```diff
@@ -1,7 +1,15 @@
 from django.contrib.auth.models import User, Group
 from rest_framework import viewsets
 from .serializers import UserSerializer, GroupSerializer
-
```
Please keep two lines blank before the begin of `class UserViewSet` ([PEP8](https://www.python.org/dev/peps/pep-0008/))

nils-wisiol commented on api/boilerplate/views.py:
```diff
@@ -17,3 +25,50 @@ class GroupViewSet(viewsets.ModelViewSet):
     queryset = Group.objects.all()
     serializer_class = GroupSerializer
+
+# This function is based on django.contrib.auth.views.login
```
For function comments, please use `"""` (cf [PEP257](https://www.python.org/dev/peps/pep-0257/))

The [current (1.11.4) version](https://github.com/django/django/blob/1.11.4/django/contrib/auth/views.py#L136) of `django.contrib.auth.views.login` is just a stub that appears to be depreciated and superseded by a [class-based view](https://github.com/django/django/blob/1.11.4/django/contrib/auth/views.py#L67). Please clarify which function your code is based on.

nils-wisiol commented on api/entrypoint.sh:
```diff
@@ -4,7 +4,7 @@
 host=db; port=3306; n=120; i=0; while ! (echo > /dev/tcp/$host/$port) 2> /dev/null; do [[ $i -eq $n ]] && >&2 echo "$host:$port not up after $n seconds, exiting" && exit 1; echo "waiting for $host:$port to come up"; sleep 1; i=$((i+1)); done
 
 # collect static files
-python manage.py collectstatic --clear --noinput || exit 1
```
The `collectstatic` command is designed to move static files from their locations in the different Django modules into a common directory. Afaik, it is not best practice to include static files into the common directory yourself. Please move the static files into their [canonical location](https://docs.djangoproject.com/en/1.11/ref/contrib/staticfiles/).

Also, keep in mind that you're adding duplicate code here. Much of the static files added here was already added in earlier commits (cf #25 )

nils-wisiol reviewed (changes requested):
Please make sure to include e2e for the functionality you add with this PR.

nils-wisiol commented:
(I changed the PR base to make an 'early' review possible. This should nevertheless be merged to `master`, not into `add_open_id_provider`.)

larissazech committed:
Moves static files into a common directory

larissazech committed:
Moves static files into a common directory

BenjaminKeller committed:
added interface between eID-Service and eID-Server

Armagetron committed:
changed testing framework to chakram

BenjaminKeller committed:
added interface between eID-Service and OpenId-Provider

auvin committed:
keep two blank lines (PEP8)

auvin committed:
Added python scripts to travis to test client registration

Before the test container starts a client is registered by the provider
and a django-user is created.
After running the test container, client and user will be deleted.

auvin committed:
Django app should not load itself

    - removed 'boilerplate' out of apps list
    - adjusted path for templates

larissazech committed:
Fix indentation and Code Style and add function comment

larissazech committed:
Fix indentation and Code Style and add function comment

larissazech commented:
Committed minor code style changes. 
Still missing: e2e tests.
Regarding issue #62, we propose to make massive changes on this code base. Therefore, the pull request has to wait until those changes are done and e2e tests are implemented.

BenjaminKeller committed:
adapted client checker to pass GET parameter through

nils-wisiol commented:
is this currently waiting for review?

BenjaminKeller commented:
No, we have to made changes according to #62. Then, we will call for review.

larissazech committed:
wip

m273d15 committed:
modifiy auth view to redirecto to eService init page

larissazech committed:
wip; untested as branch eidservice_integration is not yet merged into master

BenjaminKeller commented on api/boilerplate/views.py:
```diff
@@ -17,3 +27,50 @@ class GroupViewSet(viewsets.ModelViewSet):
+    elif request.method == "POST":
+        params = request.POST
+    
+    return HttpResponseRedirect(redirect_to + '?' + params.urlencode())
```
I wouldn't do it that way. You are hardcoding the hostname. Change
```
redirect_to = '/api/eIdService/init'
```
This works and you don't have to specify the hostname.

larissazech replied:
Thank you very much for the review! Done and tested, works fine!

BenjaminKeller commented on api/boilerplate/views.py:
```diff
@@ -17,3 +27,50 @@ class GroupViewSet(viewsets.ModelViewSet):
+    if request.method == "GET":
+        params = request.GET
+    elif request.method == "POST":
+        params = request.POST
```
Does the openId Protocol not specify if the request has to be made by POST or GET?

larissazech replied:
> Authorization Servers MUST support the use of the HTTP GET and POST methods defined in RFC 2616 [RFC2616] at the Authorization Endpoint. Clients MAY use the HTTP GET or POST methods to send the Authorization Request to the Authorization Server. If using the HTTP GET method, the request parameters are serialized using URI Query String Serialization, per Section 13.1. If using the HTTP POST method, the request parameters are serialized using Form Serialization, per Section 13.2.

taken from section 3.1.2.1 of [The OpenID Connect Core 1.0 Spec](http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)

BenjaminKeller replied:
OK, great.

BenjaminKeller reviewed (commented):
@larissazech Could delete all stuff which isn't needed? (ClientChecker etc.) I added two comments.

BenjaminKeller commented:
OK, great!

Von: larissazech<mailto:notifications@github.com>
Gesendet: Mittwoch, 30. August 2017 08:32
An: swp-fu-eid/eid-fu-swp<mailto:eid-fu-swp@noreply.github.com>
Cc: BenjaminKeller<mailto:benjamin-keller@live.de>; Comment<mailto:comment@noreply.github.com>
Betreff: Re: [swp-fu-eid/eid-fu-swp] Adjust open id provider login (#48)


@larissazech commented on this pull request.

________________________________

In api/boilerplate/views.py<https://github.com/swp-fu-eid/eid-fu-swp/pull/48#discussion_r135980839>:

> @@ -17,3 +27,50 @@ class GroupViewSet(viewsets.ModelViewSet):
     """
     queryset = Group.objects.all()
     serializer_class = GroupSerializer
+
+def open_id_auth(request):
+
+    redirect_to='https://eid.local/api/eIdService/init'
+    params = None
+    if request.method == "GET":
+        params = request.GET
+    elif request.method == "POST":
+        params = request.POST


Authorization Servers MUST support the use of the HTTP GET and POST methods defined in RFC 2616 [RFC2616] at the Authorization Endpoint. Clients MAY use the HTTP GET or POST methods to send the Authorization Request to the Authorization Server. If using the HTTP GET method, the request parameters are serialized using URI Query String Serialization, per Section 13.1. If using the HTTP POST method, the request parameters are serialized using Form Serialization, per Section 13.2.

taken from section 3.1.2.1 of The OpenID Connect Core 1.0 Spec<http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/swp-fu-eid/eid-fu-swp/pull/48#discussion_r135980839>, or mute the thread<https://github.com/notifications/unsubscribe-auth/Ac76WxUS4icr69ovOQK2YWDb6BFg6x9wks5sdQH7gaJpZM4O6WnR>.

larissazech committed:
Fix minor merge error

larissazech committed:
Change hard coded hostname for redirect to relative url

larissazech committed:
Remove test for oidc login page as there is no user-pw-login anymore

auvin commented:
Referring to #41, this is closed.
